synfig-core/src/modules/mod_ffmpeg/trgt_ffmpeg.cpp

```diff
@@ -178,9 +184,51 @@ ffmpeg_trgt::init(ProgressCallback* cb = nullptr)
 	// Windows always have ffmpeg
 	ffmpeg_binary_path = synfig::OS::get_binary_path().parent_path() / filesystem::Path("ffmpeg.exe");
 	if (!FileSystemNative::instance()->is_file(ffmpeg_binary_path.u8string())) {
-		synfig::error("Expected FFmpeg binary not found: %s", ffmpeg_binary_path.u8_str());
-		ffmpeg_binary_path.clear();
-	}
+        	synfig::warning("FFmpeg binary not found at expected path: %s, application will try to look at other paths", ffmpeg_binary_path.u8_str());
+        	ffmpeg_binary_path.clear();
+    	}
+
+	if (ffmpeg_binary_path.empty()) {
+
+        	DWORD size = GetEnvironmentVariableW(L"PATH", nullptr, 0);
+        	if (size != 0) {
+
+            		std::vector<wchar_t> pathBuffer(size);
+            		GetEnvironmentVariableW(L"PATH", pathBuffer.data(), size); // filling pathBuffer with the value of "PATH" ENVIRONMENT variable
+
+            		// Convert buffer to std::wstring
+            		std::wstring pathVariable(pathBuffer.data());
+
+            		std::wstringstream ws(pathVariable);
+            		std::wstring dir;
+            		constexpr wchar_t ENV_PATH_SEPARATOR = ';';
```

Comment:
Ugh...
Could you try convert to UTF-8 earlier and then using regular string only?
You probably could do it by using synfig::filesystem::Path::from_native() and later using its regular (utf8) methods. On Windows, it should parse the native string as UTF-16...

Reply:
Should i convert `pathVariable` to utf-8 or `dir` to utf-8?

Reply:
Pathbuffer

Reply:
Pathbuffer is not a single path, it contains all the path stored in the Environment path variable as a string separated by semicolon `;` 

That's why I think I need to write my own utf-8 converter code,
Because I don't think it's good  to use filesystem (which I guess is made for handing individual paths) to just convert to utf-8 

Which way should I go?

Reply:
As `synfig::filesystem::Path` just a lexical handle of path (it doesn't check if file/folder/path actually exists, etc. - it just handle the string by itself), I think it is okay to use it as a trick now for this conversion.


@ice0 ?